LUCI/LuciFit.py

```diff
@@ -542,40 +558,63 @@ def calculate_params(self):
-        else:
-            cons = sigma_cons + vel_cons + vel_cons_multiple
+        if self.freeze is False:  # Not freezing velocity and broadening
+            initial = np.ones((3 * self.line_num + 1))  # Initialize solution vector  (3*num_lines plus continuum)
+            initial[-1] = self.cont_estimate(sigma_level=2)  # Add continuum constant and initialize it
+            lines_fit = []  # List of lines which already have been set up for fits
+            for mod in range(self.line_num):  # Step through each line
+                lines_fit.append(self.lines[mod])  # Add to list of lines fit
+                amp_est, vel_est, sigma_est = self.line_vals_estimate(self.lines[mod])  # Estimate initial values
+                initial[3 * mod] = amp_est - initial[-1]  # Subtract continuum estimate from amplitude estimate
+                initial[3 * mod + 1] = vel_est  # Set wavenumber
+                initial[3 * mod + 2] = sigma_est  # Set sigma
+            # Set constraints
+            sigma_cons = self.sigma_constraints()  # Call sigma constraints
+            vel_cons = self.vel_constraints()  # Call velocity constraints
+            vel_cons_multiple = self.multiple_component_vel_constraint()
+            # CONSTRAINTS
+            if 'NII6548' in self.lines and 'NII6583' in self.lines and self.nii_cons is True:  # Add additional constraint on NII doublet relative amplitudes
+                nii_constraints = self.NII_constraints()
+                cons = sigma_cons + vel_cons + vel_cons_multiple + nii_constraints
+            else:
+                cons = sigma_cons + vel_cons + vel_cons_multiple
+            soln = minimize(nll, initial,
+                            method='SLSQP',
+                            options={'disp': False, 'maxiter': 30},
+                            tol=1e-2,
+                            args=(), constraints=cons
+                            )
+        else:  # Freezing velocity and broadening
+            initial = np.ones((self.line_num + 1))  # Initialize solution vector  (3*num_lines plus continuum)
+            initial[-1] = self.cont_estimate(sigma_level=2)  # Add continuum constant and initialize it
+            lines_fit = []  # List of lines which already have been set up for fits
+            for mod in range(self.line_num):  # Step through each line
+                lines_fit.append(self.lines[mod])  # Add to list of lines fit
+                amp_est, vel_est, sigma_est = self.line_vals_estimate(self.lines[mod])  # Estimate initial values
+                initial[mod] = amp_est - initial[-1]  # Subtract continuum estimate from amplitude estimate
+                initial_positions[mod] = vel_est
+                initial_sigmas[mod] = sigma_est
+            soln = minimize(nll, initial,
+                            method='SLSQP',
+                            options={'disp': False, 'maxiter': 30},
+                            tol=1e-2,
+                            args=())
```

Comment:
the minimize could be out of the if/else clause and have constraints=[] for the else case

Reply:
I tried that but constraints being empty was throwing errors.